Sources/Opus/Opus.Custom.swift

```diff
@@ -0,0 +1,266 @@
+import AVFoundation
+import Copus
+import Copuswrapper
+import Foundation
+
+public extension Opus {
+	///
+	/// Implements a custom opus encoder / decoder.
+	/// Custom implementations can have non-standard frame sizes
+	///
+	final class Custom {
+		private let opusCustomMode: OpaquePointer
+		let encoder: OpaquePointer
+		let decoder: OpaquePointer
+		private let format: AVAudioFormat
+		public let frameSize: Int32
+
+		public init(format: AVAudioFormat,
+		            application _: Application = .audio,
+		            frameSize: UInt32 = 128) throws
+		{
+			if !format.isValidOpusPCMFormat {
+				throw Opus.Error.badArgument
+			}
+			self.format = format
+			self.frameSize = Int32(frameSize)
+
+			var error: Opus.Error = .ok
+
+			// Create custom parameters
+			guard let customMode = opus_custom_mode_create(
+				Int32(format.sampleRate),
+				Int32(frameSize),
+				&error.rawValue
+			) else { throw error }
+			opusCustomMode = customMode
+
+			// Create custom encoder
+			guard let opusEncoder = opus_custom_encoder_create(
+				customMode,
+				Int32(format.channelCount),
+				&error.rawValue
+			) else { throw error }
+
+			encoder = opusEncoder
+
+			// Create custom decoder
+			guard let opusDecoder = opus_custom_decoder_create(
+				customMode,
+				Int32(format.channelCount),
+				&error.rawValue
+			) else { throw error }
+			decoder = opusDecoder
+		}
+
+		deinit {
+			opus_encoder_destroy(encoder)
+			opus_decoder_destroy(decoder)
+			opus_custom_mode_destroy(opusCustomMode)
+		}
+
+		///
+		/// Wrapper onto the opus_custom_encoder_ctl function
+		/// - Parameter request The Opus CTL to change
+		/// - Parameter value The value to set it to
+		///
+		/// - Returns Opus.Error code raw type
+		public func encoderCtl(request: Int32, value: Int32) -> Opus.Error.RawValue {
```

Comment:
What’s the purpose of this method?

Second, why does it return an `Opus.Error.RawValue` rather than return (or throw) an `Opus.Error`?

Reply:
Is there something I can add to the documentation of the method? I thought I'd explained the why pretty clearly.
The underlying opus documentation is here: https://www.opus-codec.org/docs/opus_api-1.3.1/group__opus__custom.html#gacf3d854e681cae3b693ee56b54459226 with a list of them here:
https://www.opus-codec.org/docs/opus_api-1.3.1/group__opus__encoderctls.html

yes, it probably could return the Opus.Error directly. I'll make that change.